operator/internal/controller/redpanda/redpanda_controller.go

```diff
@@ -185,8 +185,9 @@ func (r *RedpandaReconciler) Reconcile(ctx context.Context, req ctrl.Request) (r
 	// so that we can immediately calculate cluster status
 	// from and sync in any subsequent operation that
 	// early returns
+	restartOnConfigChange := rp.Annotations != nil && rp.Annotations[RestartClusterOnConfigChangeKey] == "true"
```

Comment:
FYI: this will conflict (minorly) with https://github.com/redpanda-data/redpanda-operator/pull/1003

Reply:
👍 I can wait until you merge that and then update this accordingly